src/core/sock.c

```diff
@@ -313,6 +314,11 @@ static int nn_sock_setopt_inner (struct nn_sock *self, int level,
                 return -EINVAL;
             dst = &self->rcvbuf;
             break;
+        case NN_RCVMAXSIZE:
+            if (nn_slow (val <= -1))
```

Comment:
I might tend to accept anything as <=-1 as acceptable, then changing documentation to say "any negative integer disables the limit"